k-distribution/include/builtin/domains.k

```diff
@@ -860,6 +861,12 @@ module K-REFLECTION-SYMBOLIC [symbolic]
   syntax Bool ::= #isVariable(K) [function, hook(KREFLECTION.isVariable)]
 endmodule
 
+module META
+  imports STRING-SYNTAX
+  // Takes as input a string and returns a K term
+  syntax KItem ::= #parseAST(String) [function, hook(META.parseAST), impure]
```

Comment:
Three things. First, this is not an impure function. Second, we already have a parse function in K-IO. It's not currently implemented in any backend, so my suggestion would be to delete it, but it does need to be deleted if you want this to be the new signature of parsing in K. Third, this should be poly(0).